projects/igniteui-angular/migrations/common/tsUtils.ts

```diff
@@ -264,6 +264,30 @@ export const getTypeDefinitionAtPosition =
         if (definition.kind.toString() === 'method') {
             return getMethodTypeDefinition(langServ, definition);
         }
+        if (entryPath.endsWith('.ts')) {
+            // for ts files we can use the type checker to look up a specific node
+            // and attempt to resolve its actual type
+            const sourceFile = langServ.getProgram().getSourceFile(entryPath);
+            const typeChecker = langServ.getProgram().getTypeChecker();
+            // const node = (tss as any).getTouchingPropertyName(sourceFile, position); -> tss internal that looks up a node
+            const node = findNodeAtPosition(sourceFile, position);
+            const symbolFromNode = typeChecker.getSymbolAtLocation(node)
+            const nodeType = typeChecker.getTypeOfSymbolAtLocation(symbolFromNode, node)
+            if (nodeType) {
+                const typeArguments = typeChecker.getTypeArguments(nodeType as tss.TypeReference);
+                if (typeArguments && typeArguments.length < 1) {
+                    // it's not a generic type so try to look up its name and fileName
+                    // atm we do not support migrating union/intersection generic types
+                    // a type symbol (type) should have only one declaration
+                    // if the type is 'any' or 'some', there will be no type symbol
+                    const name = nodeType.getSymbol()?.getName();
+                    const fileName = nodeType.getSymbol()?.getDeclarations()[0].getSourceFile().fileName;
```

Comment:
`getDeclarations()` can return undefined btw, not sure if it can return empty array